bc_obps/compliance/service/penalty_calculation_service.py

```diff
@@ -211,17 +299,26 @@ def calculate_penalty(
                 compliance_obligation=obligation,
                 fee_date=date.today(),
                 accrual_start_date=accrual_start_date.strftime("%Y-%m-%d"),  # type: ignore[union-attr]
+                accrual_final_date=final_accrual_date,
+                accrual_frequency=CompliancePenalty.AccrualFrequency.DAILY,
+                is_compounding=True,
                 penalty_type=CompliancePenalty.PenaltyType.AUTOMATIC_OVERDUE,
             )
 
         last_calculation_day = final_accrual_date if final_accrual_date else cls.TODAY
 
+        submission_date = cls._get_last_supplementary_submission_date(obligation)
+        if submission_date is not None:
+            effective_start_date = submission_date + timedelta(days=31)
+        else:
+            effective_start_date = invoice.due_date + timedelta(days=1)
+
```

Comment:
Is this block accounting for the large penalty on an unpaid supplementary version and changing the start date of the large penalty? I'm not sure this logic supports the need here. Let's chat about this one